nsxt/resource_nsxt_ip_pool_allocation_ip_address_test.go

```diff
@@ -0,0 +1,140 @@
+/* Copyright © 2018 VMware, Inc. All Rights Reserved.
+   SPDX-License-Identifier: MPL-2.0 */
+
+package nsxt
+
+import (
+	"fmt"
+	"github.com/hashicorp/terraform/helper/resource"
+	"github.com/hashicorp/terraform/terraform"
+	"github.com/vmware/go-vmware-nsxt"
+	"math/rand"
+	"net/http"
+	"testing"
+	"time"
+)
+
+const (
+	// how often to check destroyment of allocated ip address
+	destroyTestPercent = 10
+
+	// how long to wait for updated allocation IP address list (to check destroyment)
+	waitSeconds = 120
+)
+
+func TestAccResourceNsxtIpPoolAllocationIPAddress_basic(t *testing.T) {
+	poolName := getIpPoolName()
+	poolResourceName := "data.nsxt_ip_pool.acceptance_test"
+	resourceName := "nsxt_ip_pool_allocation_ip_address.test"
+
+	resource.Test(t, resource.TestCase{
+		PreCheck:  func() { testAccPreCheck(t) },
+		Providers: testAccProviders,
+		CheckDestroy: func(state *terraform.State) error {
+			return testAccNSXIpPoolAllocationIPAddressCheckDestroy(state, poolResourceName, resourceName)
+		},
+		Steps: []resource.TestStep{
+			{
+				Config: testAccNSXIpPoolAllocationIPAddressCreateTemplate(poolName),
```

Comment:
I notice this test is failing with 409 (conflict) during create. Hope to provide more details on this later.

Reply:
I don't see this issue and have no clue about its cause. Please provide some suggestions how to deal with it.

Reply:
Thanks for the changes. 409 Conflict was due to my pool not having IP to allocate, so this is fine.